### ISSUE TITLE
TypeLoadException from Logger.Create when running DotNetOpenAuth under mono

### ISSUE THREAD

**Issue by AArnott** [DotNetOpenAuth/DotNetOpenAuth#151]
From [StackOverflow report](http://stackoverflow.com/questions/10805773/error-when-pushing-dotnetopenauth-to-staging-or-production-environment).

An exception was thrown by the type initializer for DotNetOpenAuth.Reporting
InnerException: System.TypeInitializationException: An exception was thrown by the type initializer for DotNetOpenAuth.Logger ---> System.TypeLoadException: A type                             load exception has occurred.
  at DotNetOpenAuth.Logger.Create (System.String name) [0x00000] in <filename unknown>:0
  at DotNetOpenAuth.Logger.CreateWithBanner (System.String name) [0x00000] in <filename unknown>:0
  at DotNetOpenAuth.Logger..cctor () [0x00000] in <filename unknown>:0
  --- End of inner exception stack trace ---
  at DotNetOpenAuth.Reporting.Initialize () [0x00000] in <filename unknown>:0
  at DotNetOpenAuth.Reporting.set_Enabled (Boolean value) [0x00000] in <filename unknown>:0
  at DotNetOpenAuth.Reporting..cctor () [0x00000] in <filename unknown>:0
StackTrace:   at DotNetOpenAuth.OpenId.RelyingParty.OpenIdRelyingParty..ctor (ICryptoKeyStore cryptoKeyStore, INonceStore nonceStore) [0x00000] in <filename unknow                            n>:0
  at DotNetOpenAuth.OpenId.RelyingParty.OpenIdRelyingParty..ctor (IOpenIdApplicationStore applicationStore) [0x00000] in <filename unknown>:0
  at DotNetOpenAuth.OpenId.RelyingParty.OpenIdRelyingParty..ctor () [0x00000] in <filename unknown>:0

**Comment by capdragon** [comment]
Andrew, thanks for adding this fix. But I somehow get the exact same error still (as you posted above). Do we need to instead try to catch System.TypeInitializationException?

**Comment by AArnott** [comment]
Let's give it a try.

**Comment by AArnott** [comment]
@capdragon can you please try [this one](http://sdrv.ms/JBK1xf)

**Comment by capdragon** [comment]
Same error... very strange. 

I've moved my production environment to a Windows hosting solution and of course it works great there. This is where i'll keep it since DotNetOpenAuth works there. However, I still have the mono environment up and a branch of my app that I could use to keep testing this if you'd like me too. If this is fixed, I would move back to the mono environment.

**Comment by AArnott** [comment]
Curious whether you have log4net.dll in your application's bin directory,
and if so, what exact version?

Thanks.

## 

Andrew Arnott
"I [may] not agree with what you have to say, but I'll defend to the death
your right to say it." - S. G. Tallentyre

On Sat, Jun 2, 2012 at 10:41 AM, CaptDragon <
reply@reply.github.com

**Comment by capdragon** [comment]
I did not have log4net.dll. I put it in there and guess what? Yep, it worked. Now i'm getting another error but it has nothing to do with this error. So I've actually decided to stay on a Windows Production environment as I'm finding it too difficult to debug problems once I push it on to the Mono testing  and production servers which are different than my windows development environment. Go ahead and close this puppy down! Thanks for your help Andrew.
